### PR TITLE
Upgrade to TS 3.8

### LINKED ISSUE #718: Add support for TypeScript 3.8.x and potentially 3.9.x
**Enhancement** 
We should update framework to make sure it works with current `latest` which is 3.8.3. We should also consider testing with the `next` version of 3.9.x which should be shipping soon.

### PR DESCRIPTION
**Pull request by maier49**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [x] Unit or Functional tests are included in the PR

**Description:**

Resolves #718

### CODE REVIEW

File: src/stores/process.ts [hunk 1]
```diff
@@ -321,7 +320,8 @@ export function processExecutor<T = any, P extends object = DefaultPayload>(
 					});
 
 					if (isThenable(result)) {
-						return result.then((result) => {
+						result.then();
```
Comment (agubler) [inline src/stores/process.ts:323]: Is this supposed to be here?

File: package.json [hunk 2]
```diff
@@ -86,7 +86,7 @@
     "rimraf": "~2.6.2",
     "selenium-webdriver": "3.6.0",
     "sinon": "~4.1.3",
-    "typescript": "3.4.5"
+    "typescript": "3.8.3"
```
Comment (agubler) [inline package.json:89]: Do we want to change the version, we are mostly trying to make the package forwards compatible. Is that possible?

File: src/stores/process.ts (hunk 1 above)
Comment (maier49) [reply src/stores/process.ts]: Nope, removed

File: package.json (hunk 2 above)
Comment (maier49) [reply package.json]: Oh that was just my misunderstanding. I've reverted the package version, still builds fine against 3.4.5

**Comment by maier49** [comment]
This worked with the `next` version of TypeScript as well with no further changes needed than for 3.8

**Comment by maier49** [comment]
@agubler Ok so now if we see a file using `registerRouterInjector` and none of the imports will match the route regex (determined by resolving the import relative to the directory of the current file) then it logs out a warning.

**Pull request merged**